mlrun/datastore/model_providers.py

```diff
@@ -0,0 +1,229 @@
+# Copyright 2025 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import warnings
+from abc import ABC
+from collections.abc import Awaitable
+from typing import Callable, Optional, TypeVar
+
+import mlrun
+from mlrun.datastore.abstract_base import (
+    BaseRemoteClient,
+)
+
+T = TypeVar("T")
+
+
+class ModelProvider(BaseRemoteClient, ABC):
```

Comment:
Why inherit ABC? No abstract interfaces defined, and also BaseRemoteClient inherits from ABC